lib/TextOperations.cpp

```diff
@@ -199,6 +199,44 @@ size_t TextOperations::getUnicodeCharactersCount(const std::string & text)
 	return conv.from_bytes(text).size(); 
 }
 
+std::string TextOperations::shortenUnicodeString(const std::string & input, int maxLength)
+{
+    int txtlen = getUnicodeCharactersCount(input);
+    std::string trimmedString = input;
+    trimRightUnicode(trimmedString, std::max(0, txtlen-maxLength));
+
+    return trimmedString;
+}
+
+std::string TextOperations::concatenateStringsWithDelimiter(const std::vector<std::string>& strings, const std::string& delimiter)
+{
+    std::string result;
+    
+    for(const std::string& str : strings) {
+        result += str + delimiter;
+    }
+
+    if(!result.empty() && result.size() >= delimiter.size()) {
+        result.erase(result.end() - delimiter.size(), result.end());
+    }
+
+    return result;

```

Comment:
Something strange: why not iterating over iterators and skip adding delimiter for the last one?

Reply:
changed as suggested.